rules-reviewed/eap7/eap6/hsearch.windup.xml

```diff
@@ -0,0 +1,1197 @@
+                </hint>
+            </perform>
+        </rule>
+        <!-- org.hibernate.search.query.engine.spi.HSQuery.getExtendedSearchIntegrator() -->
+        <!-- should be at most SearchIntegrator, preferably removed altogether -->
+        <rule id="hsearch-00216">
+            <when>
+                <javaclass references="org.hibernate.search.query.engine.spi.HSQuery.getExtendedSearchIntegrator{*}">
+                    <location>METHOD_CALL</location>
+                </javaclass>
+            </when>
+            <perform>
+                <hint title="Deprecated method HSQuery#getExtendedSearchIntegrator in Hibernate Search 5" effort="1" severity="mandatory">
+                    <message>should be at most SearchIntegrator, preferably removed altogether</message>
+                    <link href="http://hibernate.org/search/documentation/migrate/5.5/#api-changes" title="Hibernate Search 5.4 to 5.5 migration" />
+                    <link
+                        href="https://access.qa.redhat.com/documentation/en/red-hat-jboss-enterprise-application-platform/version-7.0/migration-guide/#migrate_hibernate_search_advanced_integrator_changes"
+                        title="Advanced Integrator Changes in Hibernate Search 5.x" />
+                    <tag>hibernate-search</tag>
+                </hint>
+            </perform>
+        </rule>
+        <!-- org.hibernate.search.engine.spi.DocumentBuilderIndexedEntity.getFieldCacheOption() -->
+        <rule id="hsearch-00217">
+            <when>
+                <javaclass references="org.hibernate.search.engine.spi.DocumentBuilderIndexedEntity.getFieldCacheOption{*}">
+                    <location>METHOD_CALL</location>
+                </javaclass>
+            </when>
+            <perform>
+                <hint title="Deprecated method DocumentBuilderIndexedEntity#getFieldCacheOption in Hibernate Search 5" effort="1" severity="mandatory">
+                    <message>The `DocumentBuilderIndexedEntity#getFieldCacheOption()` method has been deprecated. There is no replacement.</message>
+                    <link href="http://hibernate.org/search/documentation/migrate/5.5/#api-changes" title="Hibernate Search 5.4 to 5.5 migration" />
+                    <link
+                        href="https://access.qa.redhat.com/documentation/en/red-hat-jboss-enterprise-application-platform/version-7.0/migration-guide/#migrate_hibernate_search_advanced_integrator_changes"
+                        title="Advanced Integrator Changes in Hibernate Search 5.x" />
+                    <tag>hibernate-search</tag>
+                </hint>
+            </perform>
+        </rule>
+        <!-- org.hibernate.search.spi.BuildContext.getIndexingStrategy() -->
+        <!-- Scheduled for removal. Use BuildContext.getIndexingMode() instead. -->
+        <rule id="hsearch-00218">
+            <when>
+                <javaclass references="org.hibernate.search.spi.BuildContext.getIndexingStrategy({*}">
```

Comment:
Why does this `getIndexingStrategy({*}` have the opening paren? Other rules for deprecated methods don't have that.